src/identical-placeholders.js

```diff
@@ -0,0 +1,184 @@
+const { parse } = require('@formatjs/icu-messageformat-parser');
+const set = require('lodash.set');
+const get = require('lodash.get');
+const diff = require('jest-diff');
+const requireNoCache = require('./util/require-no-cache');
+const getTranslationFileSource = require('./util/get-translation-file-source');
+const deepForOwn = require('./util/deep-for-own');
+
+const sortAstNodes = (a, b) => `${a.type}${a.value}`.localeCompare(`${b.type}${b.value}`);
+
+const compareAst = (astA, astB) => {
+  // Skip raw text
+  const astAFiltered = astA.filter(a => a.type !== 0).sort(sortAstNodes);
```

Comment:
Could you reference the types [by enum name](https://github.com/formatjs/formatjs/blob/main/packages/icu-messageformat-parser/types.ts#L8) instead? That should make it more forwards compatible and understandable. It looks like you should be able to import `TYPES` from `@formatjs/icu-messageformat-parser` and use references like `TYPES.literal` instead of 0.